### PR TITLE
Add no-sandbox to ImportedDynamicFrameworkProcessor

### PR DESCRIPTION
**Pull request by keith**
Partially fixes https://github.com/bazelbuild/rules_apple/issues/746

### CODE REVIEW

**Comment by keith** [comment]
@brentleyjones @thii can you test?

File: apple/internal/partials/framework_import.bzl [hunk 1]
```diff
@@ -165,6 +165,7 @@ def _framework_import_partial_impl(ctx, targets, targets_to_avoid):
             outputs = [framework_zip],
             arguments = [args],
             mnemonic = "ImportedDynamicFrameworkProcessor",
+            execution_requirements = {"no-sandbox": "1"},
```
Comment (brentleyjones) [inline apple/internal/partials/framework_import.bzl:168]: This _works_, but we have it conditionally add in `if provisioning_profile:` instead. that way we only jump out of the sandbox when we are signing.

Comment (keith) [reply apple/internal/partials/framework_import.bzl:168]: Hrm interesting. I wonder what the ramifications are, if any, around conditionally executing in the sandbox. I've never seen that pattern before.

Comment (keith) [reply apple/internal/partials/framework_import.bzl]: Updated with that behavior

**Pull request merged**